in_toto/keylib.go

```diff
@@ -517,23 +548,48 @@ func VerifySignature(key Key, sig Signature, unverified []byte) error {
 		if !ok {
 			return ErrKeyKeyTypeMismatch
 		}
-		switch key.Scheme {
-		// TODO: add support for more hash algorithms
-		case ecdsaSha2nistp256:
-			hashed := sha256.Sum256(unverified)
-			// Unmarshal the ASN.1 DER marshalled ecdsa signature to
-			// ecdsaSignature. asn1.Unmarshal returns the rest and an error
-			// we can skip the rest here..
-			_, err = asn1.Unmarshal(sigBytes, &ecdsaSignature)
-			if err != nil {
-				return err
+		curveSize := parsedKey.(*ecdsa.PublicKey).Curve.Params().BitSize
+		var hashed []byte
+		// implement https://tools.ietf.org/html/rfc5656#section-6.2.1
+		// We determine the curve size and choose the correct hashing
+		// method based on the curveSize
+		switch {
+		case curveSize <= 256:
+			if err := matchEcdsaScheme(curveSize, key.Scheme); err != nil {
+				return ErrCurveSizeSchemeMismatch
+			}
+			if _, ok := hashMapping["sha256"]; !ok {
+				return ErrUnsupportedHashAlgorithm
+			}
+			hashed = hashToHex(hashMapping["sha256"](), unverified)
+		case 256 < curveSize && curveSize <= 384:
+			if err := matchEcdsaScheme(curveSize, key.Scheme); err != nil {
+				return ErrCurveSizeSchemeMismatch
+			}
+			if _, ok := hashMapping["sha384"]; !ok {
+				return ErrUnsupportedHashAlgorithm
+			}
+			hashed = hashToHex(hashMapping["sha384"](), unverified)
+		case curveSize > 384:
+			if err := matchEcdsaScheme(curveSize, key.Scheme); err != nil {
+				return ErrCurveSizeSchemeMismatch
 			}
-			if err := ecdsa.Verify(parsedKey.(*ecdsa.PublicKey), hashed[:], ecdsaSignature.R, ecdsaSignature.S); err == false {
-				return ErrInvalidSignature
+			if _, ok := hashMapping["sha512"]; !ok {
+				return ErrUnsupportedHashAlgorithm
 			}
+			hashed = hashToHex(hashMapping["sha512"](), unverified)
 		default:
-			// supported key schemes will get checked in validateKey
-			panic("unexpected Error in GenerateSignature function")
+			panic("unexpected Error in VerifySignature function")
```

Comment:
Oh, that must have slipped through in https://github.com/in-toto/in-toto-golang/pull/56. Thanks for fixing!